app/src/main/java/ch/epfl/sdp/contamination/ConcreteAnalysis.java

```diff
@@ -4,112 +4,161 @@
+                        contactDuration.put(person.getKey(), timeCloseBy);
+                        break;
+                    default:
+                        // Do nothing: this carrier does not affect my status
+                }
+            }
+
+            return contactDuration;
+        }
+
+        @Override
+        public void updateInfectionPredictions(Location location, Date startTime, Callback<Void> callback) {
+
+            Date now = new Date(System.currentTimeMillis());
+            receiver.getUserNearbyDuring(location, startTime, now, aroundMe -> {
+                modelInfectionEvolution(identifySuspectContacts(aroundMe));
+                callback.onCallback(null);
+            });
+
+            //Method 1 : (synchrone)
+//        int badMeetings = receiver.getAndResetSickNeighbors(me.getUniqueId());
+//        updateCarrierInfectionProbability(me.getIllnessProbability() + badMeetings*TRANSMISSION_FACTOR);
+
+            //method 2 : (asynchrone)
+            receiver.getNumberOfSickNeighbors(me.getUniqueId(), res -> {
+                float badMeetings = 0;
+                if(!((Map)(res)).isEmpty()){
+                    badMeetings =  ((float) (((HashMap) (res)).get(publicAlertAttribute)));
+                }
+                updateCarrierInfectionProbability(me.getIllnessProbability() + badMeetings * TRANSMISSION_FACTOR);
+                cachedSender.resetSickAlerts(me.getUniqueId());
+            });
+        }
+
+        @Override
+        public Carrier getCarrier() {
+            return me;
+        }
+
+        @Override
+        public boolean updateStatus(InfectionStatus stat) {
+            if(stat != me.getInfectionStatus()){
+                float previousIllnessProbability = me.getIllnessProbability();
+                me.evolveInfection(stat);
+                if(stat == InfectionStatus.INFECTED) {
+                    //Now, retrieve all user that have been nearby the last UNINTENTIONAL_CONTAGION_TIME milliseconds
+
+                    //1: retrieve your own last positions
+                    SortedMap<Date,Location> lastPositions = cachedSender.getLastPositions();
+
+                    //2: Ask firebase who was there
+
+                    Set<String> userIds = new HashSet<>();
+                    lastPositions.forEach((date,location) -> receiver.getUserNearby(location,date,around->{
+                        around.forEach(neighbor -> {
+                            if(neighbor.getInfectionStatus()!= InfectionStatus.INFECTED){ // only non-infected users need to be informed
+                                userIds.add(neighbor.getUniqueId()); //won't add someone already in the set
+                            }
+                        });
+                    }));
+                    //Tell those user that they have been close to you
+                    //TODO: discuss whether considering only the previous Illness probability is good
```

Comment:
That's fine to do that, apart from the case when you discover that you are infected after less than UNINTENTIONAL_CONTAGION_TIME. In that case you should pass prob=1

Reply:
I thought that during the previous process, i.e. before what I added with this branch, when you are identifiying suspected Persons you met, you consider also the ones that have a probability greater than one. Thus, if now one one these persons gets sick the next day and alerts you, you will be calculating the infectedProbability with a coefficient of (previousProba + 1) if you do not do what is specified by your comment. In fact, we already considered the day before that you had a suspicious meeting.
The goal of all of this is that when you met someone infected you are not more alerted than what you should be. The "biggest" alert is when a person was completely healthy and now becomes sick, than you should be notified 100%
Hope it makes it clear.

Reply:
Thus I propose to let it as I wrote it ^^

Reply:
Agree, let see how that works first